### DISCUSSION TITLE
Is `autoSelectFamily` possible to counteract Node.js 18 dns.lookup changes?

### DISCUSSION THREAD

**Discussion by MikeMcC399** [sindresorhus/got#2235] [Q&A] [answered]
Is it possible to use `autoSelectFamily` from [socket.connect(options[, connectListener])](https://nodejs.org/dist/latest-v18.x/docs/api/net.html#socketconnectoptions-connectlistener) of Node.js 18 and later to check if a development server is running or not, when the hostname is `localhost` and you don't know beforehand if it is listening on the IPv4 `127.0.0.1` loopback address or on the IPv6 `::1` loopback address?

This is turning out to be a problem with some dev servers, for instance Angular or Vite, as GitHub has now migrated their runners to use Node.js 18 as default and the change in logic in Node.js has kicked in. [Node.js 17.0.0](https://github.com/nodejs/node/releases/tag/v17.0.0) introduced the change in https://github.com/nodejs/node/commit/1b2749ecbe "(SEMVER-MAJOR) dns: default to verbatim=true in dns.lookup()". (See History in [dns.lookup(hostname[, options], callback)](https://nodejs.org/dist/latest-v18.x/docs/api/dns.html#dnslookuphostname-options-callback).)

**Comment by sindresorhus** [comment, marked as answer]
Got already has a solution for this. You can use the `dnsLookupIpVersion` option to force IPv4:

```js
await got('http://localhost:3000', {
	dnsLookupIpVersion: 4
});
```

Alternatively, use 127.0.0.1 directly instead of localhost.

The autoSelectFamily option could be added as a pass-through to Node.js, but given the existing workarounds, I don't think it's necessary.
